src/ImageSharp/Common/Helpers/Shuffle/IComponentShuffle.cs

```diff
@@ -97,15 +99,15 @@ public void RunFallbackShuffle(ReadOnlySpan<byte> source, Span<byte> dest)
         {
             ref uint sBase = ref Unsafe.As<byte, uint>(ref MemoryMarshal.GetReference(source));
             ref uint dBase = ref Unsafe.As<byte, uint>(ref MemoryMarshal.GetReference(dest));
-            int n = source.Length / 4;
+            nuint n = (nuint)(uint)source.Length / 4;
```

Comment:
If you wonder why these casts were suggested: it's to tell the JIT that it's unsigned, so it can emit a bit-shift for the division by 4, instead of the expensive integer division.

Reply:
With the nint usage now, the intermediate cast to uint is removed.

Reply:
Hm, now I can't follow.

Here it should be `nuint` (note the **u** for unsigned), as [codegen](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdvHgwBiECAAo2MbABMA8l0kBPWWOxcADzA3hgwAHwMuP5cAJQMALyR0QEsADIwXADmGAAWDAD0DCga1OYWDBxWGLb25E4uHl6+McGhEVEx8UkMDlU8sSlc6Zk5+UUlpuY6DFzVDABC2FANbp4+fgFtYcldiZEOcwNDI9l5hcWl5TNHNUtQ9c5rzZtBITudAd0Ht7F91YMYqcxhdJmVprouP07tgAF6rJobVrvDpDb69KEAk4ZM7jS5TMw3aGLOGPRrrFpbFG7L77DHQv4M7Gjc4TUoAXyAA==) is best. The intermediate cast to `uint` removes the `movsxd` (which I don't like 😉).

For `Unsafe.Add` both `nint` and `nuint` produce equal code. Hence go here with `nuint`.

Reply:
The issue is with the combination of these two lines:

`nuint n = (nuint)(uint)span.Length / 4;`
`uint packed = Unsafe.Add(ref sBase, n);`

This all looks good on SharpIo, but when I compile it, it complains about the type of `n` in the second line. `Unsafe.Add` requires its second argument to be of type `int`. Hence the cast I put in earlier, which puts in the `movsxd` instruction again.

Please advise on the entire method.

Reply:
> `Unsafe.Add` requires its second argument to be of type `int`

There are [overloads](https://apisof.net/catalog/c9442b39-7483-70da-b3b4-817e3281a1ed) for `nint` and `nuint` too. Note: `IntPtr` <-> `nint` and `UIntPtr` <-> `nuint` for the overloads, as these map transparently.

Reply:
Builds still need to support .NET Standard 1.3 which [doesn't seem to have them](https://apisof.net/catalog/c9442b39-7483-70da-b3b4-817e3281a1ed).

Maybe we can remove 1.3 in ImageSharp 2.0, I'm not aware of any significant platforms where upgrading to netstandard 2.0 is a significant blocker.  @JimBobSquarePants  thoughts?

Reply:
Yeah kill it. Compatibility chart looks good to me

Reply:
> still need to support .NET Standard 1.3

Ah that's the point I missed. Sorry for not remembering this.
If support for NS 1.3 is needed for ImageSharp 2.0 too, would it make sense to have a shim for missing methods?

Reply:
> Yeah kill it.

Though it's too big scope in scope for this PR I guess. There should be a separate PR, this one will be blocked on if that's acceptable. I wonder who can take it, my next item in the queue is #1884.

Reply:
I can do it easily enough.

Reply:
I agree that a too big a change for here. 

What do we do with IComponentShuffle in **_this_** PR ?

Reply:
If you can wait 24 hours I can get the target framework removal merged in and you can avoid any hackery

Reply:
OK... 3 hours. 1.3 is now gone!

Reply:
OK, great, than we can go with the `nuint` version of IComponentShuffle.
Checked it on [sharplab.io](https://sharplab.io/#gist:bae34a10615c1574892da98bf3e476af) and the generated machine code looks good to me!

Reply:
That link still uses the cast no? `nuint` directly is better.
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdHXjwYBZABSwAZgys3cAITww0DLm4wGXgBKY1NzCICGMWwwAGsYABMGAF4GAFUuXGwnQQBBRMTHGBdPb18QjWoIiOIAdmjYhMSqiIBfGjagA=

Reply:
Nope, the compiler assumes the `int` overload than and throws an error.

Reply:
Removing netstandard 1.3 was to facilitate the use of the overloads. Are we now saying they are not available?

Reply:
So.... It looks like you need version 6 for this to work. Which is unsupported in .NET Core 2.1

Reply:
The [docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.add?view=net-6.0) are also not very clear. They speak about _.NET Platform Extensions_, but even those versions do not support the UIntPtr overload (which we would like to use to avoid the `movsxd` machine instruction).

The presence of a NuGet version, doesn't mean it has all the overloads, right? NS 2.0 might still have a reduced API and still miss the overload we're looking for.

If this is the case, I feel we're better off using `int` as loop variables. Let me know what you think is the best way forward here.

Reply:
I agree. Let’s leave it as int for now

Reply:
OK, done that.